server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -246,25 +257,23 @@ private void validateArguments(CreationMessageEntry entry, MailboxSession sessio
         attachmentChecker.assertAttachmentsExist(entry, session);
     }
 
-    @VisibleForTesting void assertIsUserOwnerOfMailboxes(CreationMessageEntry entry, MailboxSession session) throws MailboxNotOwnedException {
-        if (containsMailboxNotOwn(entry.getValue().getMailboxIds(), session)) {
+    @VisibleForTesting void assertIsUserOwnerOfMailboxes(List<MailboxId> mailboxIds, MailboxSession session) throws MailboxNotOwnedException {
+        if (containsMailboxNotOwn(mailboxIds, session)) {
             throw new MailboxNotOwnedException();
         }
     }
 
-    private boolean containsMailboxNotOwn(List<String> mailboxIds, MailboxSession session) {
+    private boolean containsMailboxNotOwn(List<MailboxId> mailboxIds, MailboxSession session) {
```

Comment:
I think it will more understandable to make the revert function "allMailboxOwned" and to use the negation of it.
This bring more understandable naming.
But maybe it just me

Reply:
Looks cool